docs/generalized_distribution.md

```diff
@@ -0,0 +1,83 @@
+# Generalized distribution
+
+As of Zeebe version 8.3 we have introduced a generic way to distribute commands to other partitions.
+We felt the need to generalize this as we are seeing more feature requests that will require
+distribution.
+
+## Why do we need to distribute commands?
+
+Zeebe is a distributed system. We have multiple partitions working together to achieve high
+availability and performance.
+
+When we deploy a new process this process must be available on all
+partitions. If this is not the case we would only be able to start instances of this process on a
+single partition. By distributing it instances can be started across all partitions.
+
+## How do we distribute commands?
+
+On a high-level we use commands and event to distribute commands. The image below provides a nice
+overview of the flow of commands and events that are written during distribution.
+
+![Generalized distribution overview](assets/generalized_distribution.png)
```

Comment:
🔧 Let's make this note more helpful by adding how they can check this (`TypedRecord.isCommandDistributed()`)
> [..] so the processor should have some checks built-in to prevent this from happening.